lisa/tools/kdump.py

```diff
@@ -0,0 +1,206 @@
+from typing import List, Type, cast
+
+from assertpy import assert_that
+
+from lisa.executable import Tool
+from lisa.operating_system import Debian, Posix, Redhat, Suse
+from lisa.tools import Cat, Kexec, Makedumpfile, Sed
+from lisa.tools.service import Systemctl
+from lisa.util import LisaException
+
+
+class Kdump(Tool):
+    """
+    kdump is a feature of the Linux kernel that creates crash dumps in the event of a
+    kernel crash. When triggered, kdump exports a memory image (also known as vmcore)
+    that can be analyzed for the purposes of debugging and determining the cause of a
+    crash.
+
+    kdump tool manages the kdump feature of the Linux kernel. Different distro os has
+    different kdump tool.
+    """
+
+    # If the file /sys/kernel/kexec_crash_loaded does not exist. This means that the
+    # currently running kernel either was not configured to support kdump, or that a
+    # crashkernel= commandline parameter was not used when the currently running kernel
+    # booted. Value "1" means crash kernel is loaded, otherwise not loaded.
+    #
+    # It also has /sys/kernel/kexec_crash_size file, which record the crash kernel size
+    # of memory reserved. We don't need to check this file in our test case.
+    kexec_crash = "/sys/kernel/kexec_crash_loaded"
+
+    # This file shows you the current map of the system's memory for each physical
+    # device. We can check /proc/iomem file for memory reserved for crash kernel.
+    # grep -i Crash /proc/iomem
+    iomem = "/proc/iomem"
+
+    # Following are the configuration setting required for system and dump-capture
+    # kernels for enabling kdump support.
+    required_kernel_config = [
+        "CONFIG_KEXEC",
+        "CONFIG_CRASH_DUMP",
+        "CONFIG_PROC_VMCORE",
+    ]
+
+    dump_path = "/var/crash"
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Kexec, Makedumpfile]
+
+    @property
+    def command(self) -> str:
+        if isinstance(self.node.os, Redhat):
+            return "kdumpctl"
+        elif isinstance(self.node.os, Debian):
+            return "kdump-config"
+        elif isinstance(self.node.os, Suse):
+            return "kdumptool"
+        else:
+            posix_os: Posix = cast(Posix, self.node.os)
+            raise LisaException(
+                "Please check the accurate kdump tool command for distro "
+                f"{posix_os.name} {posix_os.information.version}."
+            )
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def _install(self) -> bool:
+        if isinstance(self.node.os, Redhat):
+            self.node.os.install_packages("kexec-tools")
+        elif isinstance(self.node.os, Debian):
+            self.node.os.install_packages("linux-crashdump")
+        elif isinstance(self.node.os, Suse):
+            self.node.os.install_packages("kdump")
+
+        return self._check_exists()
+
+    def check_kernel_config(self, config_path: str) -> None:
```

Comment:
The method name should align with what's checked, like `check_required_kernel_config`

Reply:
Done